core/templates/dev/head/exploration_editor/preview_tab/PreviewTab.js

```diff
@@ -42,17 +123,21 @@ oppia.controller('PreviewTab', [
       $timeout(function() {
         oppiaPlayerService.populateExploration({
           category: explorationCategoryService.savedMemento,
-          init_state_name: explorationInitStateNameService.savedMemento,
+          init_state_name: stateName,
           param_changes: explorationParamChangesService.savedMemento,
           param_specs: explorationParamSpecsService.savedMemento,
           states: explorationStatesService.getStates(),
           title: explorationTitleService.savedMemento,
           skin_customizations: {
             panels_contents: explorationGadgetsService.getPanelsContents()
           }
-        }, []);
+        }, manualParamChanges);
         $scope.isExplorationPopulated = true;
       }, 200);
+    };
+
+    $scope.$on('updateActiveState', function(evt, stateName) {
```

Comment:
Could you explain again why this is here / why it is needed? It doesn't seem to tie into anything in particular so I'm wondering if all the updateActiveState event-passing is still necessary.